nixos/modules/services/web-apps/cryptpad.nix

```diff
@@ -0,0 +1,213 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.cryptpad;
+
+  inherit (lib) strings mdDoc mkOption types mkIf mkMerge;
+
+  # The Cryptpad configuration file isn't JSON, but a JavaScript source file that assigns a JSON value
+  # to a variable.
+  cryptpadConfigFile = builtins.toFile "cryptpad_config.js" ''
+    module.exports = ${builtins.toJSON cfg.config}
+  '';
+
+  # Derive domain names for Nginx configuration from Cryptpad configuration
+  mainDomain = strings.removePrefix "https://" cfg.config.httpUnsafeOrigin;
+  sandboxDomain = if isNull cfg.config.httpSafeOrigin then mainDomain else strings.removePrefix "https://" cfg.config.httpSafeOrigin;
+
+in {
+  options.services.cryptpad = {
+    enable = lib.mkEnableOption "cryptpad";
+
+    package = lib.mkPackageOption pkgs "cryptpad" {};
+
+    configureNginx = mkOption {
+      description = mdDoc ''
+        Configure Nginx as a reverse proxy for Cryptpad.
+        Note that this makes some assumptions on your setup, and sets settings that will
+        affect other virtualHosts running on your Nginx instance, if any.
+        Alternatively you can configure a reverse-proxy of your choice.
+      '';
+      type = types.bool;
+      default = false;
+    };
+
+    confinement = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        FIXME: Enables 'confinement' (explain what it does?)
```

Comment:
Can probably just say if you don't know what it is you don't need it -- I'll be happy to be the sole user of this as it's a rather high maintenance option..

Reply:
I would love to know what this is though :)
Is this some chroot kind of stuff? Is this something you've added or does this also come from upstream Cryptpad documentation?

Reply:
`systemd-confinement` is specific to NixOS, where it's a module adding `systemd.services.${service}.confinement.*` options to restrict the Nix store paths available at runtime to `${service}`, using `pkgs.closureInfo`.
https://discourse.nixos.org/t/nixos-policy-regarding-systemd-confinement/18976